src/ore/src/metrics.rs

```diff
@@ -97,6 +94,94 @@ pub struct MetricsRegistry {
     third_party: Registry,
 }
 
+/// A wrapper for metrics to require delete on drop semantics
+#[derive(Clone)]
+pub struct DeleteOnDropWrapper<M> {
```

Comment:
It took me a bit to understand what this does, but now I think I get it: This wraps all prometheus metric types that are exported from here in `DeleteOnDropWrapper`, so anything that `use`s them by name from here gets delete-on-drop by default. Nice!

Reply:
Let me add some documentation :)